playbooks/setup.yaml

```diff
@@ -32,3 +32,21 @@
     when:
     - services['firewalld.service'] is defined
     - services['firewalld.service']['state'] == 'running'
+ 
+  - name: Create directories for Pulp
+    ansible.builtin.file:
+      path: "{{ item }}"
+      state: directory
+      mode: '0755'
+    loop:
+      - /var/lib/pulp/settings
+      - /var/lib/pulp/pulp_storage
+      - /var/lib/pulp/pgsql
+      - /var/lib/pulp/containers
+
+  - name: Create settings.py for Pulp
+    ansible.builtin.lineinfile:
+      path: /var/lib/pulp/settings/settings.py
+      line: "CONTENT_ORIGIN='http://{{ ansible_hostname }}:8080'"
+      create: yes
```

Comment:
I wonder why https://pulpproject.org/pulp-oci-images/docs/admin/tutorials/quickstart/#create-the-directories-and-settings suggests writing a `settings.py` instead of using environment variables. I recall you could also set `PULP_CONTENT_ORIGIN` as an environment variable instead to configure it but I'll check that with the Pulp people.

Reply:
Quoting @mdellweg from Matrix:

> I think that's a matter of taste. Both work. Some complex settings are easier coded in settings.py. OTOH, having executable settings always makes me shiver.

So my recommendation would be to define it in the `.container` file as an `Environment` key to simplify things. Then you can avoid the entire `settings.py` file